### ISSUE TITLE
Add support for API V3

### ISSUE THREAD

**Issue by jonnynux** [pwelch/virustotal_api#25]
By the end of the year it's necessary to add support for API V3.

**Comment by pwelch** [comment]
Notification on VT Docs: https://developers.virustotal.com/v3.0/reference#overview

**Comment by pwelch** [comment]
@jonnynux thanks as always for your help! 

I've queued up a milestone for the next version and added this issue.

**Comment by crondaemon** [comment]
Hi,

I started working on the port of the gem to V3. According to @jonnynux, our idea is to adhere more to the new API structure. That means that basically the code can stay as is, but the classes have to be renamed. `/file/report`, mapping to `FileReport` is now `/files`. In this case we would have a new class called `Files`. The API V3 basically add `/analyse` when you want to re-do something. Moreover more commands are supported by just appending them to the API (see https://developers.virustotal.com/v3.0/reference#file). So our proposal is to change the signature of `FileReport::scan` to

```ruby
namespace VirustotalAPI
  class Files < Base
    def scan(id, command)
 ```

in order to support all the commands for searching, and

```ruby
...
def upload(path)
```

for requesting a scan.

Long story short: the gem basically will be rewritten to be more similar to the API V3 than to its previous versions.

If you agree, we will start it soon.

**Comment by pwelch** [comment]
@crondaemon This all sounds okay to me! Feel free to get started. Also, if it's easier to use RSpec for the tests, I am okay switching to that.

Thank you and @jonnynux for working and helping maintain this update!
